packages/mui-joy/src/AltDateWidget/AltDateWidget.tsx

```diff
@@ -0,0 +1,274 @@
+import { MouseEvent, memo, useCallback, useEffect, useReducer, useState } from 'react';
+import {
+  ariaDescribedByIds,
+  parseDateString,
+  toDateString,
+  pad,
+  DateObject,
+  FormContextType,
+  RJSFSchema,
+  StrictRJSFSchema,
+  TranslatableString,
+  WidgetProps,
+} from '@rjsf/utils';
+import Stack from '@mui/joy/Stack';
+import Button from '@mui/joy/Button';
+
+function rangeOptions(start: number, stop: number) {
+  const options = [];
+  for (let i = start; i <= stop; i++) {
+    options.push({ value: i, label: pad(i, 2) });
+  }
+  return options;
+}
+
+function readyForChange(state: DateObject) {
+  return Object.values(state).every((value) => value !== -1);
+}
+
+function dateElementProps(
```

Comment:
#4004 implements a function to replace these. Once it is merged can you update this. Thanks!

Reply:
ok i will update when it willl be merged